src/Microsoft.AspNetCore.Authentication/Events/FailureContext.cs

```diff
@@ -9,10 +9,14 @@ namespace Microsoft.AspNetCore.Authentication
     /// <summary>
     /// Provides failure context information to handler providers.
     /// </summary>
-    public class FailureContext : BaseControlContext
+    public class FailureContext : HandleRequestContext<RemoteAuthenticationOptions>
```

Comment:
Yeah we know its not amazing, this https://github.com/aspnet/Security/issues/711 is tracking making it less crappy